src/Symfony/Component/DependencyInjection/Compiler/AttributeAutoconfigurationPass.php

```diff
@@ -35,21 +77,77 @@ protected function processValue($value, bool $isRoot = false)
             || !$value->isAutoconfigured()
             || $value->isAbstract()
             || $value->hasTag('container.ignore_attributes')
-            || !($reflector = $this->container->getReflectionClass($value->getClass(), false))
+            || !($classReflector = $this->container->getReflectionClass($value->getClass(), false))
         ) {
             return parent::processValue($value, $isRoot);
         }
 
-        $autoconfiguredAttributes = $this->container->getAutoconfiguredAttributes();
         $instanceof = $value->getInstanceofConditionals();
-        $conditionals = $instanceof[$reflector->getName()] ?? new ChildDefinition('');
-        foreach ($reflector->getAttributes() as $attribute) {
-            if ($configurator = $autoconfiguredAttributes[$attribute->getName()] ?? null) {
-                $configurator($conditionals, $attribute->newInstance(), $reflector);
+        $conditionals = $instanceof[$classReflector->getName()] ?? new ChildDefinition('');
+
+        if ($this->classAttributeConfigurators) {
+            foreach ($classReflector->getAttributes() as $attribute) {
+                if ($configurator = $this->classAttributeConfigurators[$attribute->getName()] ?? null) {
+                    $configurator($conditionals, $attribute->newInstance(), $classReflector);
+                }
             }
         }
-        if (!isset($instanceof[$reflector->getName()]) && new ChildDefinition('') != $conditionals) {
-            $instanceof[$reflector->getName()] = $conditionals;
+
+        if ($this->methodAttributeConfigurators || $this->parameterAttributeConfigurators) {
+            $constructorReflector = $this->getConstructor($value, false);
+            if ($constructorReflector) {
```

Comment:
`if ($this->parameterAttributeConfigurators && $constructorReflector = $this->getConstructor($value, false)) {`

Reply:
Done, made it a bit smarter.